amethyst_input/src/bindings.rs

```diff
@@ -118,14 +204,26 @@ where
             bindings.push(bind.clone());
             self.actions.insert(id.into(), bindings);
         }
+        Ok(())
     }
 
     /// Removes an action binding that was assigned previously.
-    pub fn remove_action_binding<T: Hash + Eq + ?Sized>(&mut self, id: &T, binding: &[Button])
+    pub fn remove_action_binding<T: Hash + Eq + ?Sized>(
+        &mut self,
+        id: &T,
+        binding: &[Button],
+    ) -> Result<(), ActionRemovedError>
     where
         AC: Borrow<T>,
     {
-        let mut kill_it = false;
+        for i in 0..binding.len() {
+            for j in 0..binding.len() {
+                if i != j && binding[i] == binding[j] {
+                    return Err(ActionRemovedError::BindingContainsDuplicates);
+                }
+            }
+        }
+        let kill_it;
```

Comment:
why not `let kill_it = if ...`?

Reply:
It makes the control flow in the `else` a little wonky because the value is never used in that branch. Also, the `if` branch may or may not exit the function early.